Core/Src/main.c

```diff
@@ -45,6 +47,14 @@
 
 /* USER CODE BEGIN PV */
 
+const uint8_t MCP3000_START_BIT = 0b00000001;
+const uint8_t MCP3004_CH0 = 0b10000000;
+const uint16_t MCP3004_MAX_VALUE = 0b0000001111111111;
```

Comment:
Although totally valid, it is very hard to read... In general, if your are displaying pin configs or any "binary values" it is common practice to use hex instead of binary, functionally the same but more readable!

Reply:
Fixed in 13a6881b9bb2e8dbbc73b3124064bed00f60962b